[jit] de-optionalize SourceRange context

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#32880 [jit] de-optionalize SourceRange context**
* #32879 [jit] Delete the ErrorReport default constructor

The PR below made it impossible to construct a SourceRange without a
context, so get rid of its optional-ness

Differential Revision: [D19670923](https://our.internmc.facebook.com/intern/diff/D19670923)

suo committed:
[jit] de-optionalize SourceRange context

The PR below made it impossible to construct a SourceRange without a
context, so get rid of its optional-ness

[ghstack-poisoned]

suo committed:
Update on "[jit] de-optionalize SourceRange context"


The PR below made it impossible to construct a SourceRange without a
context, so get rid of its optional-ness

[ghstack-poisoned]

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit 46f54c31:


* **1/1** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=46f54c31022c60c3645465e124d7cb781b52c73e).

### :detective: 1 new failure recognized by patterns
The following build failures do not appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4392729) pytorch_linux_xenial_py3_6_gcc5_4_build (1/1)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=93903860) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=93903860))
<details>
<summary>
<code>Automatic merge failed; fix conflicts and then commit the result.</code>
</summary>

```
do so (now or later) by using -b with the checkout command again. Example: 
 
  git checkout -b <new-branch-name> 
 
HEAD is now at 46f54c3102 Update on "[jit] de-optionalize SourceRange context" 
+ git reset --hard 46f54c31022c60c3645465e124d7cb781b52c73e 
HEAD is now at 46f54c3102 Update on "[jit] de-optionalize SourceRange context" 
+ git merge --allow-unrelated-histories --no-edit --no-ff 7b65acdf9e8345d2ffbbc3b85ae6667f7260407e 
Auto-merging torch/csrc/jit/script/error_report.cpp 
CONFLICT (content): Merge conflict in torch/csrc/jit/script/error_report.cpp 
Automatic merge failed; fix conflicts and then commit the result. 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

suo committed:
Update on "[jit] de-optionalize SourceRange context"


The PR below made it impossible to construct a SourceRange without a
context, so get rid of its optional-ness

Differential Revision: [D19670923](https://our.internmc.facebook.com/intern/diff/D19670923)

[ghstack-poisoned]

suo committed:
Update on "[jit] de-optionalize SourceRange context"


The PR below made it impossible to construct a SourceRange without a
context, so get rid of its optional-ness

Differential Revision: [D19670923](https://our.internmc.facebook.com/intern/diff/D19670923)

[ghstack-poisoned]

suo committed:
Update on "[jit] de-optionalize SourceRange context"


The PR below made it impossible to construct a SourceRange without a
context, so get rid of its optional-ness

Differential Revision: [D19670923](https://our.internmc.facebook.com/intern/diff/D19670923)

[ghstack-poisoned]
